/*
 * This file was automatically generated by EvoSuite
 * Tue Sep 26 22:17:00 GMT 2023
 */

package org.mockito.internal;

import org.junit.Test;
import static org.junit.Assert.*;
import static org.evosuite.shaded.org.mockito.Mockito.*;
import static org.evosuite.runtime.EvoAssertions.*;
import java.lang.annotation.Annotation;
import java.util.List;
import java.util.Vector;
import org.evosuite.runtime.EvoRunner;
import org.evosuite.runtime.EvoRunnerParameters;
import org.evosuite.runtime.ViolatedAssumptionAnswer;
import org.junit.runner.RunWith;
import org.mockito.internal.MockHandler;
import org.mockito.internal.creation.MockSettingsImpl;
import org.mockito.internal.invocation.Invocation;
import org.mockito.internal.stubbing.InvocationContainerImpl;
import org.mockito.internal.verification.MockAwareVerificationMode;
import org.mockito.stubbing.Answer;
import org.mockito.stubbing.VoidMethodStubbable;
import org.mockito.stubbing.answers.ReturnsElementsOf;

@RunWith(EvoRunner.class) @EvoRunnerParameters(mockJVMNonDeterminism = true, useVFS = true, useVNET = true, resetStaticState = true, separateClassLoader = true) 
public class MockHandler_ESTest extends MockHandler_ESTest_scaffolding {

  @Test(timeout = 4000)
  public void test0()  throws Throwable  {
      MockHandler<Annotation> mockHandler0 = new MockHandler<Annotation>();
      MockHandler<Annotation> mockHandler1 = new MockHandler<Annotation>(mockHandler0);
      assertFalse(mockHandler1.equals((Object)mockHandler0));
  }

  @Test(timeout = 4000)
  public void test1()  throws Throwable  {
      MockSettingsImpl mockSettingsImpl0 = new MockSettingsImpl();
      MockHandler<Object> mockHandler0 = new MockHandler<Object>(mockSettingsImpl0);
      // Undeclared exception!
      try { 
        mockHandler0.setAnswersForStubbing((List<Answer>) null);
        fail("Expecting exception: NullPointerException");
      
      } catch(NullPointerException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         verifyException("java.util.ArrayList", e);
      }
  }

  @Test(timeout = 4000)
  public void test2()  throws Throwable  {
      MockHandler<Object> mockHandler0 = new MockHandler<Object>();
      InvocationContainerImpl invocationContainerImpl0 = (InvocationContainerImpl)mockHandler0.getInvocationContainer();
      assertFalse(invocationContainerImpl0.hasAnswersForStubbing());
  }

  @Test(timeout = 4000)
  public void test3()  throws Throwable  {
      MockSettingsImpl mockSettingsImpl0 = new MockSettingsImpl();
      MockHandler<Object> mockHandler0 = new MockHandler<Object>(mockSettingsImpl0);
      VoidMethodStubbable<Object> voidMethodStubbable0 = mockHandler0.voidMethodStubbable(mockSettingsImpl0);
      assertNotNull(voidMethodStubbable0);
  }

  @Test(timeout = 4000)
  public void test4()  throws Throwable  {
      Invocation invocation0 = mock(Invocation.class, new ViolatedAssumptionAnswer());
      MockHandler<Annotation> mockHandler0 = new MockHandler<Annotation>();
      try { 
        mockHandler0.handle(invocation0);
        fail("Expecting exception: NullPointerException");
      
      } catch(NullPointerException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         verifyException("org.mockito.internal.MockHandler", e);
      }
  }

  @Test(timeout = 4000)
  public void test5()  throws Throwable  {
      MockHandler<ReturnsElementsOf> mockHandler0 = new MockHandler<ReturnsElementsOf>();
      MockSettingsImpl mockSettingsImpl0 = mockHandler0.getMockSettings();
      MockHandler<MockAwareVerificationMode> mockHandler1 = new MockHandler<MockAwareVerificationMode>(mockSettingsImpl0);
      Vector<Integer> vector0 = new Vector<Integer>();
      ReturnsElementsOf returnsElementsOf0 = new ReturnsElementsOf(vector0);
      mockHandler1.invocationContainerImpl.addAnswerForVoidMethod(returnsElementsOf0);
      try { 
        mockHandler1.handle((Invocation) null);
        fail("Expecting exception: NullPointerException");
      
      } catch(NullPointerException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         verifyException("org.mockito.internal.invocation.InvocationMatcher", e);
      }
  }
}
